R/aaa_survival_prop.R

```diff
@@ -185,21 +189,36 @@ survival_prob_coxnet <- function(object, new_data, times, output = "surv", ...)
     object$fit,
     newx = new_x,
     newstrata = new_strata,
+    s = penalty,
     x = object$training_data$x,
     y = object$training_data$y,
     na.action = na.exclude,
     ...
   )
 
-  res <- stack_survfit(y, nrow(new_data)) %>%
-    dplyr::group_nest(.row, .key = ".pred") %>%
+  if (multi) {
+    names(y) <- penalty
+    keep_penalty <- TRUE
+    stacked_survfit <-
+      purrr::map_dfr(y, stack_survfit, n = nrow(new_data), .id = "penalty") %>%
+      dplyr::mutate(penalty = as.numeric(penalty)) %>%
+      dplyr::group_nest(.row, penalty, .key = ".pred")
+  } else {
+    keep_penalty <- FALSE
+    stacked_survfit <-
+      stack_survfit(y, nrow(new_data)) %>%
+      dplyr::group_nest(.row, .key = ".pred")
+  }
+  res <-
+    stacked_survfit %>%
     mutate(
       .pred = purrr::map(.pred, ~ dplyr::bind_rows(prob_template, .x))
```

Comment:
The implementation seems correct, but the mapped `bind_row()` feels like it takes longer than it should with larger data.

Reply:
I've turned it into #72 to take another stab at this one :)